src/sdx/pce/heuristic/heur.py

```diff
@@ -0,0 +1,274 @@
+
+    def solve(self, partition_tm):
+        partition_shape = np.shape(partition_tm)
+        graph = self.topology
+        # print("Partition_shape="+str(partition_shape))
+        # print(partition_tm)
+        final_result = 0
+        final_ordered_paths = []
+        for i in range(partition_shape[0] - 1, -1, -1):
+            # print("i="+str(i))
+            if i == 0:
+                partition = partition_tm[i][self.pad :]
+            else:
+                partition = partition_tm[i]
+            print(partition)
+            tm = matrix_to_connection(partition)
+            solver = TESolver(graph, tm, self.cost, self.objective)
+            ordered_paths = solver.solve()
+            graph = solver.update_graph(graph, ordered_paths)
+            final_result = final_result + ordered_paths.cost
+            final_ordered_paths.append(ordered_paths.connection_map)
+        return final_ordered_paths, final_result
+
+    def disjoint_path(self, connection):
+        # Prime path:solver
+        # prune the graph
+        # Backup path: solver again
+        # path translator -> dict.
+        pass
+
+
+if __name__ == "__main__":
+    parse = argparse.ArgumentParser()
+    parse.add_argument(
+        "-t",
+        dest="te_file",
+        required=False,
+        help="Input file for the connections or traiffc matrix, e.g. c connection.json. Required.",
+        type=str,
+    )
+    parse.add_argument(
+        "-n",
+        dest="topology_file",
+        required=False,
+        help="Input file for the network topology, e.g. t topology.json. Required.",
+        type=str,
+    )
+    parse.add_argument(
+        "-c", dest="c", required=False, default=0, help="Link cost definition", type=int
+    )
+    parse.add_argument(
+        "-b",
+        dest="b",
+        required=False,
+        default=0,
+        help="Objective: MinCost or Load balancing",
+        type=int,
+    )
+    parse.add_argument(
+        "-m",
+        dest="m",
+        required=False,
+        help="Number of connections in the random TE",
+        type=int,
+    )
+    parse.add_argument(
+        "-a",
+        dest="alg",
+        default=0,
+        help="Flag for different grouping heuristic algorithms, default is the linear partition",
+        type=int,
+    )
+    parse.add_argument("-g", dest="group", default=2, help="number of groups", type=int)
+    parse.add_argument(
+        "-o",
+        dest="result",
+        default="OUTPUT.txt",
+        help="Output file, e.g. o result.txt. If this option is not given, assume standard output.",
+        type=str,
+    )
+
+    parse.print_help()
+    args = parse.parse_args()
+    # result(args.te_file,args.node_name , args.result)
+
+    if args.topology_file is not None:
+        if args.te_file is not None:
+            # graph, tm = dot_file(args.topology_file, args.te_file)
+            print("Supporting dot file!")
+        else:
+            print("Missing the TE file!")
+            exit()
```

Comment:
This actually doesn't seem to handle topology file and te file:

```console
$ python src/sdx/pce/heuristic/heur.py -n src/sdx/pce/data/topologies/amlight.json -t src/sdx/pce/data/requests/test_request.json 
usage: heur.py [-h] [-t TE_FILE] [-n TOPOLOGY_FILE] [-c C] [-b B] [-m M] [-a ALG] [-g GROUP] [-o RESULT]

options:
  -h, --help        show this help message and exit
  -t TE_FILE        Input file for the connections or traiffc matrix, e.g. c connection.json. Required.
  -n TOPOLOGY_FILE  Input file for the network topology, e.g. t topology.json. Required.
  -c C              Link cost definition
  -b B              Objective: MinCost or Load balancing
  -m M              Number of connections in the random TE
  -a ALG            Flag for different grouping heuristic algorithms, default is the linear partition
  -g GROUP          number of groups
  -o RESULT         Output file, e.g. o result.txt. If this option is not given, assume standard output.
Supporting dot file!
Traceback (most recent call last):
  File "/tmp/pce/src/sdx/pce/heuristic/heur.py", line 264, in <module>
    te = TEGroupSolver(graph, tm, args.c, args.b)
                       ^^^^^
NameError: name 'graph' is not defined
```

Reply:
yes. commented it out that handles .dot files. making it better...